### ISSUE TITLE
Compatibility for libzmq3 / older Debians?

### ISSUE THREAD

**Issue by dginev** [erickt/rust-zmq#117]
Mostly an FYI issue, feel free to close right away.

I have had a crate depending on rust-zmq that has been running nicely on Debian jessie and wheezy for over a year now, and has built nicely with Travis with only adding:
```
addons:
  apt:
    packages:
      - libzmq3-dev
```

I decided to update to the latest master today, since I have been tracking the github repo, and sadly the comfy travis config is no longer supported, and neither are my old-fashioned Debian servers. Clearly updating to a 4.* libzmq from source would be one way out, but that's something I'd like to postpone as much as possible, extra complexity in setup and all.

Long story short, I wanted to share my updated Cargo.toml dependency which still works on legacy systems, I simply copied the last revision sha that was operational on Travis for me:

```
[dependencies.zmq]
git = "https://github.com/erickt/rust-zmq.git"
rev = "87a6bd2a7ce09200b6f0a5bdc08ef684c4951f21"
```

Since the crate is updated on crates.io somewhat rarely, I'll freeze the revision until an upgrade happens somewhere: OS upgrade on the servers, or added backwards compatibility to rust-zmq. Just a tip that may keep other rust-zmq users happy until the world moves forward.

**Comment by birkenfeld** [comment]
I have an update in #115 that will restore compatibility with 4.0.x and even 3.2.x.

Once this is merged, it would be great if you could verify that it still works with your crate.

**Comment by rotty** [comment]
@dginev Now that #115 has been merged, could you verify that your builds using older libzmq started working again, and close this issue if so?

**Comment by dginev** [comment]
... waiting for the nightly fix for [broken crate updates](https://github.com/rust-lang/cargo/issues/3340) to land and I'll give it a spin, thanks!

**Comment by rotty** [comment]
@dginev Any news on this issue?

**Comment by dginev** [comment]
Well, i was still waiting on [the cargo fix to land](https://github.com/rust-lang-nursery/rustup.rs/issues/844), but there seems to be a [viable workaround](https://github.com/rust-lang-nursery/rustup.rs/issues/844#issuecomment-264995387) that I managed to get to work for now.

I have tried the latest rust-zmq HEAD (commit be88b5f4e2dbbcabbc187d45cee2cd7c506cf541 ) and can confirm this issue is resolved - both on travis and my Debian servers.

Thanks!!
